Assignment5/GenericsHomework/Node.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace GenericsHomework
+{
+    public class Node<T>
+    {
+        public Node<T> Next { get; private set; }
+        T Value { get; set; }
+
+        public Node(T value, Node<T> next)
+        {
+            if (value is null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+            Value = value;
+
+            if (next is null)
+            {
+                Next = this;
+            }
+            else
+            {
+                Next = next;
+                if(Next.Next == Next)
+                {
+                    Next.Next = this;
+                }
+            }
+        }
+
+        public Node(T value)
+        {
+            if (value is null)
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+            Value = value;
+            Next = this;
+        }

```

Comment:
You can chain these constructors together to avoid needing to duplicate logic. Like this:
```suggestion
        public Node(T value)
            : this(value, null)
        { }
```